### PR TITLE
BEGONIA-119: Add product information in the assessment report

### PR DESCRIPTION
**Pull request by bgubbala**
Fixes ISSUE # 
BEGONIA-119: Add product information in the assessment report

## Description
Enrich the Image Scanner Assessment report with VAC product assessment info
Moved go-client from exporter to the lib

### Before fix
```
"containers": [
                          {
                            "name": "apache",
                            "id": "docker://775fa09e82e3d3110a2cbfb43d2ede4a52efa0bd3b25c12617e8b0d3a13acc81",
                            "image": "harbor.int.app-catalog.vmware.com/catalog_governor/containers/debian-11/apache:2.4.56-debian-11-r22",
                            "imageID": "docker-pullable://harbor.int.app-catalog.vmware.com/catalog_governor/containers/debian-11/apache@sha256:086606d412ac04b75ddfecb606421cfb9046b7a063b848494b88e40f9523d7c8",
                            "isInit": false,
                          }
                        ]
```

### After fix
```
"containers": [
                          {
                            "name": "apache",
                            "id": "docker://775fa09e82e3d3110a2cbfb43d2ede4a52efa0bd3b25c12617e8b0d3a13acc81",
                            "image": "harbor.int.app-catalog.vmware.com/catalog_governor/containers/debian-11/apache:2.4.56-debian-11-r22",
                            "imageID": "docker-pullable://harbor.int.app-catalog.vmware.com/catalog_governor/containers/debian-11/apache@sha256:086606d412ac04b75ddfecb606421cfb9046b7a063b848494b88e40f9523d7c8",
                            "isInit": false,
                            "vacAccessment": {
                              "trusted": true,
                              "name": "Apache",
                              "branch": "2.4",
                              "version": "2.4.56",
                              "revision": "22",
                              "releasedAt": "2023-03-31T16:13:25Z",
                              "lastVersionReleased": "2.4.56-r25",
                              "status": "ACTIVE"
                            }
                          }
                        ]
```

### CODE REVIEW

File: cnsi-inspector/pkg/image-scanner/controller.go [hunk 1]
```diff
@@ -143,6 +230,12 @@ func (c *controller) Run(ctx context.Context, policy *v1alpha1.InspectionPolicy)
 		NamespaceAssessments: make([]*itypes.NamespaceAssessment, 0),
 	}
 
+	// GetVacAdapter
+	vacAdapter, err := c.GetVacAdapter(setting)
```
Comment (snootan) [inline cnsi-inspector/pkg/image-scanner/controller.go:234]: I think this should be called only if VacAssessment is enabled.

**Review by snootan — CHANGES_REQUESTED** [review]

**Comment by snootan** [comment]
@bgubbala, I could see go-client and mocks files path changes from governor to lib folder is changed. Did we decide to do like this?

File: cnsi-inspector/pkg/image-scanner/controller.go (hunk 1 above)
Comment (bgubbala) [reply cnsi-inspector/pkg/image-scanner/controller.go:234]: it is inline with other adapter implementation(harbor)
Adapter will generate the Assessment. 

This is not the VacAssessment, it is just adapter to fetch the VacAssessment

**Comment by bgubbala** [comment]
> 
It makes more sense for them to be part of lib/ rather than in exporter.
Now it will be used by both cnsi-exporter & cnsi-inspector.

File: cnsi-inspector/pkg/image-scanner/controller.go (hunk 1 above)
Comment (bgubbala) [reply cnsi-inspector/pkg/image-scanner/controller.go:234]: True, it should have AssessmentEnabled check

**Comment by snootan** [comment]
> > 
> 
> It makes more sense for them to be part of lib/ rather than in exporter. Now it will be used by both cnsi-exporter & cnsi-inspector.

Great, Add in description as well.

**Review by snootan — CHANGES_REQUESTED** [review]

File: cnsi-inspector/pkg/image-scanner/controller.go [hunk 2]
```diff
@@ -186,6 +279,18 @@ func (c *controller) Run(ctx context.Context, policy *v1alpha1.InspectionPolicy)
 
 			// Do assessment for each workload container and also calculate the overall result.
 			for _, ct := range containers {
+
+				// Get VacAssessment
+				if policy.Spec.VacAssessmentEnabled {
+					productInfo, err := vacAdapter.GetVacProductInfo(ct.ImageID)
+					if err != nil {
+						log.Error("VAC Product Assessment Failed")
```
[the remark this answers is not in the document]
Comment (bgubbala) [reply cnsi-inspector/pkg/image-scanner/controller.go:290]: Log with imageId was already added in the vacAdapter

**Review by 4everming — APPROVED** [review]
LGTM

**Review by snootan — CHANGES_REQUESTED** [review]

**Review by snootan — APPROVED** [review]

**Pull request merged**